qgispluginci/parameters.py

```diff
@@ -228,6 +242,16 @@ def __init__(self, definition: Dict[str, Any]):
             )
         self.repository_url = get_metadata("repository")
 
+    @staticmethod
+    def validate_args(args: Namespace):
+        if args.release_version and not any(
+            re.match(pattern, args.release_version)
+            for pattern in Parameters.release_version_patterns.values()
+        ):
+            raise ValueError(
+                f"Unable to validate the release version '{args.release_version}'. Please use a version name that looks like this: 'v1.1.1', 'v1.1', '1.0.1', '1.1'."
```

Comment:
I am not sure we correctly support the prefix `v` (not checked). Have you tested it?

Reply:
`qgis-plug-ci` is able to `package` release versions such as `v0.1`. Whether this is correct depends on who you ask, but if you ask GitHub, I think the answer is 'yes' since you can use such tags both when doing doing a GitHub Release and when manually tagging commits.

Reply:
We have some tests with a `v` prefix : https://github.com/opengisch/qgis-plugin-ci/blob/master/test/test_changelog.py#L82

So I would say it's OK.

Reply:
my concern is not the changelog but rather QGIS plugin repo. We just need to be sure the `v` is dropped from the zip name and possibly in the metadata. 

I see 3 approaches:
1) we need this, so we make sure it works
2) we don't need it and want to play it safe: we don't advertise it, wait and see (or we don't even support it in the regex)
3) we don't really care, let this in and wait and see

My choice would be 2, but not strong opinion.

Reply:
FYI, there are quite a few plugins having a `v` prefix, either in the ZIP name or in their metadata on the QGIS repository.

Random one https://plugins.qgis.org/plugins/qgis_server_render_geojson/ :)

Here in the QGIS plugin manager : https://github.com/qgis/QGIS/blob/0bef833103724b65b2151afc3bfe63db4c0a6fbf/python/pyplugin_installer/version_compare.py#L60

But indeed, I prefer a proper semver tag as ell ;-)

Reply:
See changes from commits here below. In hindsight I think it's to strict to raise an exception on faulty release version patterns (NB: this PR doesn't touch the `--release_tag` arg, which I think doesn't need validation anyway). So I added a flag for people who want to bypass it. There's still a warning when a release version matches no pattern.

As for the patterns, I think it's reasonable to warn whenever a release version doesn't match the semvar pattern, whether validation has been disabled or not.